src/databricks/labs/ucx/install.py

```diff
@@ -269,27 +256,27 @@ def replace_config(self, **changes: Any):
         Persist the list of workspaces where UCX is successfully installed in the config
         """
         try:
-            config = self._installation.load(WorkspaceConfig)
+            config = self.installation.load(WorkspaceConfig)
             new_config = dataclasses.replace(config, **changes)
-            self._installation.save(new_config)
+            self.installation.save(new_config)
         except (PermissionDenied, NotFound, ValueError):
             logger.warning(f"Failed to replace config for {self._ws.config.host}")
 
     def _apply_upgrades(self):
         try:
-            upgrades = Upgrades(self._product_info, self._installation)
+            upgrades = Upgrades(self.product_info, self.installation)
             upgrades.apply(self._ws)
         except (InvalidParameterValue, NotFound) as err:
             logger.warning(f"Installed version is too old: {err}")
 
     def _configure_new_installation(self, default_config: WorkspaceConfig | None = None) -> WorkspaceConfig:
         if default_config is None:
             default_config = self._prompt_for_new_installation()
-        HiveMetastoreLineageEnabler(self._ws).apply(self._prompts, self._is_account_install)
+        HiveMetastoreLineageEnabler(self._ws).apply(self.prompts, self._is_account_install)
         self._check_inventory_database_exists(default_config.inventory_database)
         warehouse_id = self._configure_warehouse()
-
-        policy_id, instance_profile, spark_conf_dict, instance_pool_id = self._policy_installer.create(
+        policy_installer = ClusterPolicyInstaller(self.installation, self.workspace_client, self.prompts)
```

Comment:
creating `@cached_property` of ClusterPolicyInstaller will simplify the testing.

Reply:
done